arcgis-android-toolkit/src/main/java/com/esri/arcgisruntime/toolkit/ar/ArcGISArView.kt

```diff
@@ -559,9 +568,22 @@ class ArcGISArView : FrameLayout, DefaultLifecycleObserver, Scene.OnUpdateListen
         // permission on Android M and above, now is a good time to ask the user for it.
         // when the permission is requested and the user responds to the request from the OS this is executed again
         // during onResume()
-        if (isUsingARCore == ARCoreUsage.YES && renderVideoFeed && !hasPermission(CAMERA_PERMISSION)) {
-            requestPermission(context as Activity, CAMERA_PERMISSION, CAMERA_PERMISSION_CODE)
-            return
+        if (isUsingARCore == ARCoreUsage.YES && !hasPermission(CAMERA_PERMISSION)) {
+            if (permissionHasBeenRequestedPreviously(CAMERA_PERMISSION) && permissionHasBeenPermanentlyDenied(
```

Comment:
Is this `if` check part of fixing https://github.com/Esri/arcgis-runtime-toolkit-android/issues/142 or does it fix yet another issue? I assume it is meant to prevent prompting for permission if the user has checked “Don’t Ask Again”?

Reply:
The previous `if` fixes #142 . This line fixes an issue if the user had selected to not show again, we'd get stuck in a loop of trying to make these checks. This now determines if the user has selected to "Don't ask again" and if so, sets an error on the View.

It might be worth removing the entire permission requesting from the View if it turns out to be too much hassle, but we'd still need to notify the user of the lib that permissions have been denied.